funcx_endpoint/funcx_endpoint/endpoint/endpoint.py

```diff
@@ -356,6 +384,20 @@ def register_endpoint(funcx_client, endpoint_name, endpoint_uuid, endpoint_dir):
                                               endpoint_uuid,
                                               endpoint_version=ENDPOINT_VERSION)
 
+    # the service will send back a message with a 'status'='error'
+    # property if something went wrong
+    if 'status' in reg_info and reg_info['status'] == 'error':
+        msg = "Endpoint registration failed."
+        if 'reason' in reg_info:
+            msg = "Endpoint registration failed. Service fail reason provided: {}".format(reg_info['reason'])
+        raise Exception(msg)
+
+    # this is a backup error handler in case an endpoint ID is not sent back
+    # from the service
+    if 'endpoint_id' not in reg_info:
+        msg = "Endpoint ID was not included in the service's registration response."
+        raise Exception(msg)
```

Comment:
I'm first checking if there is an error message above, then doing another backup check to ensure that Endpoint ID exists.